PicPi/HomeViewController.swift

```diff
@@ -7,64 +7,94 @@
+        super.viewWillAppear(animated)
+        if let ipData = KeyChain.loadKey(PI_IP_KEYCHAIN_KEY) {
+            ip = String(decoding: ipData, as: UTF8.self)
+            tableView.isHidden = false
+            tableView.reloadData()
+            
+        }
+        
+    }
+    
+    
+    // MARK: - Private
+    
+    private func setupSearchBtn() {
+        
+        //    searchBtn.frame.size = CGSize(width: 100, height: 50)
+        view.addSubview(searchBtn)
+        //add action when search button pressed
+        searchBtn.addTarget(self, action: #selector(searchAct), for: .touchUpInside)
+        
+        searchBtn.setTitle("Search", for: .normal)
+        searchBtn.sizeToFit()
+        searchBtn.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([searchBtn.centerXAnchor.constraint(equalTo: margins.centerXAnchor) ,         searchBtn.centerYAnchor.constraint(equalTo : margins.bottomAnchor, constant:  -searchBtn.frame.height)  ,searchBtn.widthAnchor.constraint(equalToConstant: 100)  ])
+        
+    }
+    
+    
+    
+    
+    @objc func searchAct(sender: UIButton!) {
+        let addFrameVC = AddFrameViewController()
+        addFrameVC.modalPresentationStyle = .fullScreen
+        self.present(addFrameVC, animated: true, completion: nil)
+    }
+    
+    private func setupTableView() {
+        tableView.dataSource = self
+        view.addSubview(tableView)
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+        
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([  tableView.topAnchor.constraint(equalTo: margins.topAnchor) ,tableView.leftAnchor.constraint(equalTo: view.leftAnchor) , tableView.rightAnchor.constraint(equalTo: view.rightAnchor), tableView.bottomAnchor.constraint(equalTo: searchBtn.topAnchor) ])
+        
+        
+    }
+}
 
-//  func collectAgenda(for date: Date) -> Promise<Agenda> {
-//    self.collectAgenda(for: date).then { agenda in
-//
-//    }.catch { error in
-//
-//    }
-//  }
 
-  // MARK: - Private
 
-  private func doSomethingPrivately() {
+extension HomeViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
+        cell.textLabel?.text = ip
+        cell.textLabel?.textAlignment = .center
+        return cell
+    }
+}
 
-  }
+extension UIViewController {
+    func hideKeyboardWhenTappedAround() {
+        let tap = UITapGestureRecognizer(target: self, action: #selector(UIViewController.dismissKeyboard))
+        tap.cancelsTouchesInView = false
+        view.addGestureRecognizer(tap)
+    }
+    
+    @objc func dismissKeyboard() {
```

Comment:
after moving this into its own file, mark this as private since it's only used by the tap gesture